spec/feature_flag_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+
+describe Bugsnag::FeatureFlag do
+  it "has a name" do
+    flag = Bugsnag::FeatureFlag.new("abc")
+
+    expect(flag.name).to eq("abc")
+    expect(flag.variant).to be_nil
+  end
+
+  it "has an optional variant" do
+    flag = Bugsnag::FeatureFlag.new("abc", "xyz")
+
+    expect(flag.name).to eq("abc")
+    expect(flag.variant).to eq("xyz")
+  end
+
+  [
+    [123, "123"],
+    [true, "true"],
+    [false, "false"],
+    [[1, 2, 3], "[1, 2, 3]"],
+    [{ a: 1, b: 2 }, "{:a=>1, :b=>2}"],
+  ].each do |variant, expected|
+    it "converts the variant to a string if given '#{variant.class}'" do
+      flag = Bugsnag::FeatureFlag.new("abc", variant)
+
+      expect(flag.name).to eq("abc")
+      expect(flag.variant).to eq(expected)
+    end
+  end
+
+  it "sets variant to 'nil' if variant cannot be converted to a string" do
+    class StringRaiser
+      def to_s
+        raise 'Oh no you do not!'
```

Comment:
:joy:

Reply:
This was actually you! I copied it from another file lol

Reply:
Haha yeah! It made me laugh to see it again unexpectedly.